src/Streamly/Data/Fold.hs

```diff
@@ -6,16 +6,15 @@
 -- Stability   : experimental
 -- Portability : GHC
 --
--- A 'Fold' is a sink or a consumer of a stream of values.  The 'Fold' type
--- consists of an accumulator and an effectful action that absorbs a value into
```

Comment:
We should not leak internal details. As the fold type keeps evolving the internal structure will change. It will be rather hard to describe the internal structure in simple terms.